src/diffusers/models/autoencoders/autoencoder_dc.py

```diff
@@ -0,0 +1,626 @@
+            y = hidden_states.repeat_interleave(self.repeats, dim=1)
+            y = F.pixel_shuffle(y, self.factor)
+            hidden_states = x + y
+        else:
+            hidden_states = x
+
+        return hidden_states
+
+
+class Encoder(nn.Module):
+    def __init__(
+        self,
+        in_channels: int,
+        latent_channels: int,
+        block_type: Union[str, Tuple[str]] = "ResBlock",
+        block_out_channels: Tuple[int] = (128, 256, 512, 512, 1024, 1024),
+        layers_per_block: Tuple[int] = (2, 2, 2, 2, 2, 2),
+        qkv_multiscales: Tuple[Tuple[int, ...], ...] = ((), (), (), (5,), (5,), (5,)),
+        downsample_block_type: str = "pixel_unshuffle",
+    ):
+        super().__init__()
+
+        num_stages = len(block_out_channels)
+
+        if isinstance(block_type, str):
+            block_type = (block_type,) * num_stages
+
+        if layers_per_block[0] > 0:
+            self.conv_in = nn.Conv2d(
+                in_channels,
+                block_out_channels[0] if layers_per_block[0] > 0 else block_out_channels[1],
+                kernel_size=3,
+                stride=1,
+                padding=1,
+            )
+        else:
+            self.conv_in = DCDownBlock2d(
+                in_channels=in_channels,
+                out_channels=block_out_channels[0] if layers_per_block[0] > 0 else block_out_channels[1],
+                downsample=downsample_block_type == "pixel_unshuffle",
+                shortcut=False,
+            )
+
+        stages = []
+        for stage_id, (width, depth) in enumerate(zip(block_out_channels, layers_per_block)):
+            stage = []
+
+            for _ in range(depth):
+                block = get_block_from_block_type(
+                    block_type[stage_id],
+                    width,
+                    width,
+                    norm_type="rms_norm",
+                    act_fn="silu",
+                    qkv_mutliscales=qkv_multiscales[stage_id],
+                )
+                stage.append(block)
+
+            if stage_id < num_stages - 1 and depth > 0:
+                downsample_block = DCDownBlock2d(
+                    in_channels=width,
+                    out_channels=block_out_channels[stage_id + 1],
+                    downsample=downsample_block_type == "pixel_unshuffle",
+                    shortcut=True,
+                )
+                stage.append(downsample_block)
+
+            stages.append(nn.Sequential(*stage))
+
+        self.stages = nn.ModuleList(stages)
+
+        self.conv_out = nn.Conv2d(block_out_channels[-1], latent_channels, 3, 1, 1)
+        self.norm_factor = 1
+        norm_in_channels = block_out_channels[-1]
+        norm_out_channels = latent_channels
+        self.norm_group_size = norm_in_channels * self.norm_factor**2 // norm_out_channels
+
+    def forward(self, hidden_states: torch.Tensor) -> torch.Tensor:
+        hidden_states = self.conv_in(hidden_states)
+        for stage in self.stages:
```

Comment:
any reason we called it stage here?
does it make sense to call it `down_blocks` to be consistent with others?

Reply:
Yes, forgot about this. Will update